hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/hapi/ContractGetBytecodeSuite.java

```diff
@@ -71,21 +78,47 @@ public boolean canRunConcurrent() {
     @HapiTest
     final HapiSpec getByteCodeWorks() {
         final var contract = "EmptyConstructor";
+        final var canonicalUsdFee = 0.05;
+        final var canonicalQueryFeeAtActiveRate = new AtomicLong();
         return HapiSpec.defaultHapiSpec("GetByteCodeWorks")
-                .given(uploadInitCode(contract), contractCreate(contract))
-                .when()
-                .then(withOpContext((spec, opLog) -> {
-                    final var getBytecode = getContractBytecode(contract).saveResultTo("contractByteCode");
-                    allRunFor(spec, getBytecode);
+                .given(
+                        cryptoCreate(CIVILIAN_PAYER).balance(ONE_HUNDRED_HBARS),
+                        uploadInitCode(contract),
+                        contractCreate(contract))
+                .when(balanceSnapshot("beforeQuery", CIVILIAN_PAYER))
+                .then(
+                        withOpContext((spec, opLog) -> {
+                            final var getBytecode = getContractBytecode(contract)
+                                    .payingWith(CIVILIAN_PAYER)
+                                    .saveResultTo("contractByteCode")
+                                    .exposingBytecodeTo(bytes -> {
+                                        canonicalQueryFeeAtActiveRate.set(
+                                                spec.ratesProvider().toTbWithActiveRates((long)
+                                                        (canonicalUsdFee * 100 * TINY_PARTS_PER_WHOLE)));
+                                        log.info(
```

Comment:
Some leftover?

Reply:
Yes probably not helpful, will delete in next PR.